src/tab/Tabs.ts

```diff
@@ -0,0 +1,259 @@
+import { TabHeader } from './TabHeader.js';
+
+/**
+ * Component that displays content in tabs.
+ *
+ * @import
+ * ```js
+ * import '@capitec/omni-components/tab';
+ * ```
+ *
+ * @example
+ * ```html
+ * <omni-tabs>
+ *  <omni-tab header='Tab 1'>
+ *   <omni-label>Tab 1</omni-label>
+ *  </omni-tab>
+ *  <omni-tab header="Tab 2">
+ *   <omni-label>Tab 2</omni-label>
+ *  </omni-tab>
+ *  <omni-tab header="Tab 3">
+ *   <omni-label>Tab 3</omni-label>
+ *  </omni-tab>
+ * </omni-tabs>
+ *```
+ *
+ * @element omni-tabs
+ *
+ * Registry of all properties defined by the component.
+ *
+ * @fires {CustomEvent<{}>} tab-select - Dispatched when one of the omni-tab(s) is selected.
+ *
+ * @slot - Content to render inside the component body.
+ * @slot header - The header slot used to slot omni-tab-header components.
+ *
+ * @cssprop --omni-tabs-tab-bar-width - Tabs tab bar width.
+ * @cssprop --omni-tabs-tab-bar-height - Tabs tab bar height.
+ * @cssprop --omni-tabs-tab-bar-border-bottom - Tabs tab bar bottom border.
+ * @cssprop --omni-tabs-tab-bar-background-color - Tabs tab bar background color.
+ *
+ * @cssprop --omni-tab-header-font-color - Tab header component font color.
+ * @cssprop --omni-tab-header-font-family - Tab header component font family.
+ * @cssprop --omni-tab-header-font-size - Tab header component font size.
+ * @cssprop --omni-tab-header-font-weight - Tab header component font weight.
+ *
+ * @cssprop --omni-tab-header-disabled-background-color - Tab header component disabled background color.
+ * @cssprop --omni-tab-header-active-font-color - Tab component header active font color.
+ *
+ * @cssprop --omni-tab-header-height - Tab header component tab height.
+ * @cssprop --omni-tab-header-min-width - Tab header component tab min width.
+ * @cssprop --omni-tab-header-max-width - Tab header component tab max width.
+ * @cssprop --omni-tab-header-margin - Tab header component tab margin.
+ *
+ * @cssprop --omni-tab-header-hover-background-color - Tab header component tab hover background.
+ *
+ * @cssprop --omni-tab-header-indicator-bar-height - Tab header component indicator bar height.
+ * @cssprop --omni-tab-header-indicator-bar-border-radius -  Tab header component indicator bar border radius.
+ * @cssprop --omni-tab-header-indicator-bar-width - Tab header component indicator bar width.
+ *
+ * @cssprop --omni-tab-header-indicator-height - Tab header component indicator height.
+ * @cssprop --omni-tab-header-indicator-color - Tab header component indicator color.
+ * @cssprop --omni-tab-header-indicator-border-radius - Tab header component indicator border radius.
+ * @cssprop --omni-tab-header-indicator-width - Tab header component indicator width.
+ */
+@customElement('omni-tabs')
+export class Tabs extends OmniElement {
+    @state() private _observer: MutationObserver | undefined;
+
+    override connectedCallback(): void {
+        super.connectedCallback();
+        this._observer = new MutationObserver((mutations) => {
```

Comment:
What is the browser support for `MutationObserver`?
Should this not be wrapped with `if("MutationObserver" in window)` to cater for older browsers?

Reply:
Browser support for most browsers dates back to 2013.